controllers/storagecluster/exporter.go

```diff
@@ -209,9 +265,515 @@ func CreateOrUpdateServiceMonitor(r *StorageClusterReconciler, instance *ocsv1.S
+			"verbs":["get","list","watch"]
+		},
+		{
+			"apiGroups":["quota.openshift.io"],
+			"resources":["clusterresourcequotas"],
+			"verbs":["get","list","watch"]
+		},
+		{
+			"apiGroups":["objectbucket.io"],
+			"resources":["objectbuckets","objectbucketclaims"],
+			"verbs":["get","list"]
+		},
+		{
+			"apiGroups":[""],
+			"resources":["secrets"],
+			"verbs":["get","list"]
+		},
+		{
+			"apiGroups":["storage.k8s.io"],
+			"resources":["storageclasses"],
+			"verbs":["get","list","watch"]
+		},
+		{
+			"apiGroups":["ocs.openshift.io"],
+			"resources":["storageconsumers","storageclusters"],
+			"verbs":["get","list","watch"]
+		}
+	]
+}`
+
+func updateMetricsExporterClusterRoles(ctx context.Context,
+	r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	currentClusterRole := new(rbacv1.ClusterRole)
+	var expectedClusterRole = new(rbacv1.ClusterRole)
+	err := json.Unmarshal([]byte(metricsExporterClusterRoleJSON), expectedClusterRole)
+	if err != nil {
+		r.Log.Error(err, "an unexpected error occurred while unmarshalling clusterrole yaml")
+		return err
+	}
+	expectedClusterRole.Name, currentClusterRole.Name = metricsExporterName, metricsExporterName
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, currentClusterRole, func() error {
+		expectedRulesLen := len(expectedClusterRole.Rules)
+		if len(currentClusterRole.Rules) != expectedRulesLen {
+			currentClusterRole.Rules = make([]rbacv1.PolicyRule, expectedRulesLen)
+		}
+		copy(currentClusterRole.Rules, expectedClusterRole.Rules)
+		return nil
+	})
+	return err
+}
+
+const expectedMetricExporterRoleJSON = `
+{
+	"apiVersion":"rbac.authorization.k8s.io/v1",
+	"kind":"Role",
+	"metadata":{"name":"ocs-metrics-svc"},
+	"rules":[
+		{
+			"apiGroups":[""],
+			"resources":["services","endpoints","pods"],
+			"verbs":["get","list","watch"]
+		}
+	]
+}
+`
+
+func createMetricsExporterRoles(ctx context.Context,
+	r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	var expectedRole = new(rbacv1.Role)
+	err := json.Unmarshal([]byte(expectedMetricExporterRoleJSON), expectedRole)
+	if err != nil {
+		r.Log.Error(err, "an unexpected error occurred while unmarshalling metrics exporter role YAML")
+		return err
+	}
+	currentRole := new(rbacv1.Role)
+	expectedRole.Name, currentRole.Name = metricsExporterRoleName, metricsExporterRoleName
+	expectedRole.Namespace, currentRole.Namespace = instance.Namespace, instance.Namespace
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, currentRole, func() error {
+		expectedRulesLen := len(expectedRole.Rules)
+		if len(currentRole.Rules) != expectedRulesLen {
+			currentRole.Rules = make([]rbacv1.PolicyRule, expectedRulesLen)
+		}
+		copy(currentRole.Rules, expectedRole.Rules)
+		return nil
+	})
+	return err
+}
+
+const expectedMetricExporterRoleBindingJSON = `
+{
+	"apiVersion":"rbac.authorization.k8s.io/v1",
+	"kind":"RoleBinding",
+	"metadata":{"name":"ocs-metrics-svc"},
```

Comment:
`RoleBinding` is namespaced resource. Add namespace.

Reply:
It is a static template. All the dynamic parts are added in the code.